### ISSUE TITLE
[photospline] Python GLAM error in 1D monodim case (Trac #1727)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1719]
Executing the cdf_fit_1d example from package-test with the Python GLAM fails with this error:

% python cdf_fit_1d.py
SPGLAM not found, falling back on Python GLAM...
Calculating spline basis...
Calculating penalty matrix...
Reticulating splines...
	Processing dimension 0
Computing iteration 1 least squares solution...
0 negative coefficients
infeasibles: 32
Traceback (most recent call last):
  File "cdf_fit_1d.py", line 34, in <module>
    table = glam.fit(bins.cumsum(),weight,[edges[1:]],knots,order,smooth,monodim=0)
  File "/Users/jkelley/Documents/ARA/raytrace/radiospline/photospline-package-test-orig/python/photospline/glam/glam.py", line 240, in fit
    result = (nnls_mockup.nnls_normal_block(F,r),)
  File "/Users/jkelley/Documents/ARA/raytrace/radiospline/photospline-package-test-orig/python/photospline/glam/nnls.py", line 238, in nnls_normal_block
    Atb_F = Atb[:,F]
IndexError: too many indices for array

It works fine with SPGLAM.  There may be a more elegant way to fix this, but this works in nnls_normal_block():

		if Aqtb.ndim == 1:
			Atb_F = Atb[F]
		else:
			Atb_F = Atb[:,F]

with a similar change for Atb[:,G].

Migrated from https://code.icecube.wisc.edu/ticket/1727
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:12:47", 
    "description": "Executing the cdf_fit_1d example from package-test with the Python GLAM fails with this error:\n\n% python cdf_fit_1d.py\nSPGLAM not found, falling back on Python GLAM...\nCalculating spline basis...\nCalculating penalty matrix...\nReticulating splines...\n\tProcessing dimension 0\nComputing iteration 1 least squares solution...\n0 negative coefficients\ninfeasibles: 32\nTraceback (most recent call last):\n  File \"cdf_fit_1d.py\", line 34, in <module>\n    table = glam.fit(bins.cumsum(),weight,[edges[1:]],knots,order,smooth,monodim=0)\n  File \"/Users/jkelley/Documents/ARA/raytrace/radiospline/photospline-package-test-orig/python/photospline/glam/glam.py\", line 240, in fit\n    result = (nnls_mockup.nnls_normal_block(F,r),)\n  File \"/Users/jkelley/Documents/ARA/raytrace/radiospline/photospline-package-test-orig/python/photospline/glam/nnls.py\", line 238, in nnls_normal_block\n    Atb_F = Atb[:,F]\nIndexError: too many indices for array\n\nIt works fine with SPGLAM.  There may be a more elegant way to fix this, but this works in nnls_normal_block():\n\n\t\tif Aqtb.ndim == 1:\n\t\t\tAtb_F = Atb[F]\n\t\telse:\n\t\t\tAtb_F = Atb[:,F]\n\nwith a similar change for Atb[:,G].\n", 
    "reporter": "jkelley", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067167842669", 
    "component": "combo reconstruction", 
    "summary": "[photospline] Python GLAM error in 1D monodim case", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2016-06-03T17:05:24", 
    "milestone": "", 
    "owner": "jkelley", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160603T17:07:27: jkelley commented:

> Stray character in there, meant Atb.ndim == 1

**Comment by icecube-trac** [comment]
Trac update at 20160603T20:45:12: 
* jkelley changed status from "new" to "closed"
* jkelley changed resolution from "" to "fixed"
* jkelley commented:

> In changeset 146799:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="146799"
> Switch to ellipsis for indexing where needed to make 1D monotonic fitting work.  Fixes #1719
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:12:47: 
* olivas changed milestone from "June 2016" to ""
* olivas commented:

> Milestone June 2016 deleted
